### PR TITLE
Add PWM Components

### PR DESCRIPTION
**Pull request by brentru**
Adding folder of PWM output components. To start,

* Generic RGB LED
* Generic Dimmable LED
* Piezo Buzzer

### CODE REVIEW

**Review by lorennorman — COMMENTED** [review]
Looks great to me as a start! I don't want to merge it yet:
- need to add the GH:A bits for it
- needs to not get imported into production until i account for the entirely new type, it may crash things

Let me know if you need it merged though, I can make it work.

**Comment by brentru** [comment]
@lorennorman  I do not need this merged yet.

**Comment by brentru** [comment]
@lorennorman The commit above adds a new field, `pwmSetting`. This field will define if the updated feed data sends either a protocol buffer message with a fixed frequency and a variable duty cycle (using this message, https://github.com/adafruit/Wippersnapper_Protobuf/blob/master/proto/wippersnapper/pwm/v1/pwm.proto#L12) or a variable frequency with a fixed duty cycle (using this message, https://github.com/adafruit/Wippersnapper_Protobuf/blob/master/proto/wippersnapper/pwm/v1/pwm.proto#L31).

**Comment by lorennorman** [comment]
Oh nice, yeah this makes sense to me.

What if instead of:
`"pwmSetting": "fixedFrequency"`
...it was:
`"frequency": "fixed"`

**Comment by brentru** [comment]
That would work! Adding in..

**Comment by lorennorman** [comment]
@brentru I've done some schema and validator updates to this PR, improved some things. At this point, it is failing legitimately due to non-square images (a new validation I've added to fit one of Tyler's design requirements). After that is resolved it will complain about the animation being too large.

Would you mind addressing these when you get a chance?

**Comment by brentru** [comment]
@lorennorman Fixed! It is currently failing on [Validate pwm Component Definition Files](https://github.com/adafruit/Wippersnapper_Components/actions/runs/3381032136/jobs/5614482958#logs)

File: components/pwm/rgb_led/definition.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+    "displayName": "RGB LED",
+    "pwmSetting": "fixedFrequency",
+    "visualization": {
+      "type": "color-picker"
```
Comment (lorennorman) [inline components/pwm/rgb_led/definition.json:5]: @brentru I see type: `color-picker` and I'm realizing we don't have that in the discussion/diagrams on Basecamp. Want to:
- ...design out how it work for me?
- ...change it to a different type?
- ...remove the component?

**Pull request merged**